### DISCUSSION TITLE
Default UUID function in Postgre

### DISCUSSION THREAD

**Discussion by Pirastrino** [prisma/prisma#6190] [Q&A] [answered]
Hi, I am new to Prisma and trying to figure few things out.

I found out that when I define an id as `id String @id @default(uuid()) @db.Uuid` it is translated into `"id" UUID NOT NULL`.
Since PostgreSQL 9.14 there is a `gen_random_uuid()` included which generates mostly used uuid v4.

Since there is no option to set `@default(uuidv4())` or `@default(uuidv1())` etc. in `schema.prisma` either, I assume that prisma generates a random uuid v4 as well on insert so it would be nice if the generated migration translates into `"id" UUID NOT NULL DEFAULT gen_random_uuid()` maybe? Ofc. I can set it manually or with a script eg. on docker-compose up but that means I have to manually alter database after any migrate reset/deploy when in a dev mode.

If anyone has some nice workaround, I'd be very thankful 👍

**Comment by Pirastrino** [comment, marked as answer]
I just found out [dbgenerated()](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#dbgenerated). 🤦‍♂️

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
